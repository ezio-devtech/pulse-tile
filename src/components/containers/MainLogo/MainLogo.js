import React from 'react';

const MainLogo = props => <div className="wrap-logo">
  <div className="logo">
    <div className="logo-icon">
      <a href="#">
        <img src="./src/assets/images/ripple-foundation-logo.png" className="img" alt="logo"/>
      </a>
    </div>
  </div>
</div>;

export default MainLogo;
