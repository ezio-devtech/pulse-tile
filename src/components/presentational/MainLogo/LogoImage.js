import React from 'react';
import image from '../../../assets/images/pulsetile-core-logo.png';

export const logo = image;