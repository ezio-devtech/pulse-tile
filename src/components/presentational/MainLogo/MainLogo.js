import React, { PureComponent } from 'react';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import { logo } from './LogoImage';

export default class MainLogo extends PureComponent {
  render() {
    const { patientsInfo, userAccount } = this.props;
    return (
      <div className="wrap-logo">
        <div className="logo">
          <div className="logo-icon">
            {userAccount.role === 'IDCR'
              ? <Link to={'/'}>
                {logo ? <img className="img" id="logo-image" alt="logo" src={logo} /> : null}
                <img className={classNames(`img logo-img ${logo ? 'hidden' : ''}`)} alt="logo" src={patientsInfo.logoB64} />
              </Link>
              : userAccount.role === 'PHR'
                ? <Link to={`/patients/${userAccount.nhsNumber}/patients-summary`}>
                        {logo ? <img className="img" id="logo-image" alt="logo" src={logo} /> : null}
                  <img className={classNames(`img logo-img ${logo ? 'hidden' : ''}`)} alt="logo" src={patientsInfo.logoB64} />
                </Link> : null }
          </div>
        </div>
      </div>
    )
  }
}
