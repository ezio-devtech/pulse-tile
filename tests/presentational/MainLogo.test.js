import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router'

import MainLogo from '../../src/components/presentational/MainLogo/MainLogo';

const patientsInfo = {
  'browserTitle': 'PulseTile',
  'logoB64': 'testLogo',
  'themeColor': 'green',
  'title': 'PulseTile',
};

const context = {};

describe('Component <MainLogo />', () => {
  it('should renders when the role of doctor', () => {
    const userAccount = { 'role': 'IDCR' };
    const tree = renderer
      .create(
        <StaticRouter location="someLocation" context={context}>
          <MainLogo
            patientsInfo={patientsInfo}
            userAccount={userAccount}
          />
        </StaticRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should renders when the role of the patient', () => {
    const userAccount = { 'role': 'PHR', 'nhsNumber': '9999999000' };
    const tree = renderer
      .create(
        <StaticRouter location="someLocation" context={context}>
          <MainLogo
            patientsInfo={patientsInfo}
            userAccount={userAccount}
          />
        </StaticRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

	it('should renders without role', () => {
		const userAccount = { 'role': '', 'nhsNumber': '9999999000' };
		const tree = renderer
			.create(
        <StaticRouter location="someLocation" context={context}>
          <MainLogo
            patientsInfo={patientsInfo}
            userAccount={userAccount}
          />
        </StaticRouter>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
