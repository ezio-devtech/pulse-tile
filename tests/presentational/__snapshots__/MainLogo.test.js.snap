// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <MainLogo /> should renders when the role of doctor 1`] = `
<div
  className="wrap-logo"
>
  <div
    className="logo"
  >
    <div
      className="logo-icon"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="img"
          src="testLogo"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`Component <MainLogo /> should renders when the role of the patient 1`] = `
<div
  className="wrap-logo"
>
  <div
    className="logo"
  >
    <div
      className="logo-icon"
    >
      <a
        href="/patients/9999999000/patients-summary"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="img"
          src="testLogo"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`Component <MainLogo /> should renders without role 1`] = `
<div
  className="wrap-logo"
>
  <div
    className="logo"
  >
    <div
      className="logo-icon"
    />
  </div>
</div>
`;
